### PR TITLE
Add AUTHORS.md with list of main contributors

### LINKED ISSUE #1489: How to give credit where it's due
We're starting to have a nice pool of contributors to the documentation, both little hands doing all the maintenance and proofreading necessary on most pages as well as people focusing on a specific area and writing lengthy tutorials.

So far we haven't given contributors much credit apart from the "and the Godot community" in the copyright statement, so there's room for improvement.

How should we do it?

An obvious first step would be to add an `AUTHORS.md` file to the repo with a list of contributors, as we have on the [code repo](https://github.com/godotengine/godot/blob/master/AUTHORS.md).

In the code repo I'm listing only contributors with over 10 commits, which is quite arbitrary, but allows to keep a reasonably well curated list of people who had some kind of role in developing Godot (excluding people who did one commit to fix a typo in the README for example). For the docs repo, such a lower limit might be too strong, as we have people who wrote thousands of words of documentation in one or two commits and would not make it to the AUTHORS file with this rule.
With the 10 commits rule we'd have only 14 authors, versus 252 with a 1 commit rule (see https://github.com/godotengine/godot-docs/graphs/contributors). With a 3 commits rule we'd be at 58.

Next to this, do we want a system to show when a given contributor wrote a whole page? On the one hand it's nice for someone who spent many hours working on a tutorial to be credited for it explicitly, on the other hand every single page was written by a contributor at some point, and after years of rewriting and improvement by other contributors, this "initial author" information would lose of its relevance. Suggestions welcome :)

### PR DESCRIPTION
**Pull request by akien-mga**
Fixes #1489.

----

I went with the "easy" approach used on the main engine repo, where we include all contributors with 10+ commits in `AUTHORS.md`.

I list them with `git shortlog -s -n -e --no-merges`, and then use the public info on their GitHub profile to decide whether to put full (public) name or only the GitHub handle.

I also included contributors who had over 10 commits on the engine repo's `doc/classes` and `modules/*/doc_classes` folders, so that class reference writers can also be credited here even if their contribution was initially on a different repository.

### CODE REVIEW

File: .mailmap [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+Frido <f.schermutzki@gmail.com>
+Hugo Locurcio <hugo.locurcio@hugo.pro> <calinou@opmbx.org>
+Hugo Locurcio <hugo.locurcio@hugo.pro> <hugo.l@openmailbox.org>
+Ignacio Etcheverry <ignalfonsore@gmail.com> <neikeq@users.noreply.github.com>
+Julian Murgia <the.straton@gmail.com>
+Kelly Thomas <kelly.thomas@hotmail.com.au>
+Leon Krause <lk@leonkrause.com>
+Leon Krause <lk@leonkrause.com> <eska@eska.me>
+Max Hilbrunner <m.hilbrunner@gmail.com>
+Max Hilbrunner <m.hilbrunner@gmail.com> <mhilbrunner@users.noreply.github.com>pn
```
Comment (mhilbrunner) [inline .mailmap:19]:
```suggestion
Max Hilbrunner <m.hilbrunner@gmail.com> <mhilbrunner@users.noreply.github.com>
```

Is this a typo? Couldn't find anything that documents a meaning for `pn` in `.mailmap`, and my entry in the main repo lacks this :)

**Comment by mhilbrunner** [comment]
Looks good to me (besides the small thing above :))

File: .mailmap (hunk 1 above)
Comment (akien-mga) [reply .mailmap]: That was to check if you do your PR reviews thoroughly ;P

Comment (mhilbrunner) [reply .mailmap]: Nah, we both know I only check *yours* thoroughly so I can nitpick endlessly :P

**Comment by mhilbrunner** [comment]
Looks even better in the new and improved version. :)

**Pull request merged**